packages/amplify_datastore/example/lib/models/Blog.dart

```diff
@@ -115,11 +126,10 @@ class Blog extends Model {
   Blog.fromJson(Map<String, dynamic> json)
       : id = json['id'],
         _name = json['name'],
-        _posts = json['posts'] is List
-            ? (json['posts'] as List)
-                .where((e) => e?['serializedData'] != null)
-                .map((e) => Post.fromJson(
-                    Map<String, dynamic>.from(e['serializedData'])))
+        _posts = json['posts'] != null
+            ? (json['posts']['items'] as List)
```

Comment:
Is `items` guaranteed to be there if `posts` is non-null?

You could add a safeguard if not:

```suggestion
            ? (json['posts']['items'] as List? ?? const [])
```

Reply:
AppSync will pass an empty `items` array when `posts` is specified in the document. When `posts` is not in the document and therefore the response too, decoding returns `null` for posts.